packages/odf/components/s3-browser/upload-objects/store.ts

```diff
@@ -0,0 +1,93 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { makeAutoObservable, toJS } from 'mobx';
+import { UploadProgress, UploadStatus } from './types';
+
+export class UploadStore {
+  uploads: Record<string, UploadProgress> = {};
+
+  constructor() {
+    makeAutoObservable(this);
+  }
+
+  // Add a file to the map
+  addFile(file: UploadProgress, key: string) {
+    this.uploads[key] = {
+      ...file,
+      uploadState: UploadStatus.INIT_STATE,
+    };
+  }
+
+  updateProgress(fileId: string, loaded: number, total: number, aborter: any) {
+    if (fileId in this.uploads) {
+      const existingData = this.uploads[fileId];
+      const update = {
+        loaded,
+        total,
+        startTime: existingData.startTime ?? Date.now(),
+      };
+      if (!existingData.abort) {
+        Object.assign(update, { abort: aborter });
+      }
+      if (
+        existingData.uploadState !== UploadStatus.UPLOAD_START &&
+        existingData.uploadState !== UploadStatus.UPLOAD_CANCELLED
+      ) {
+        Object.assign(update, { uploadState: UploadStatus.UPLOAD_START });
+      }
+      if (loaded === existingData.total) {
+        Object.assign(update, { uploadState: UploadStatus.UPLOAD_COMPLETE });
+      }
+      this.uploads[fileId] = { ...existingData, ...update };
+    }
+  }
+
+  getFile(fileId: string) {
+    return this.uploads[fileId];
+  }
+
+  uploadFailed(fileId: string) {
+    if (this.uploads[fileId]) {
+      const existingData = this.uploads[fileId];
+      const update = { uploadState: UploadStatus.UPLOAD_FAILED };
+      this.uploads[fileId] = { ...existingData, ...update };
+    }
+  }
+
+  uploadCancelled(fileId: string) {
+    if (this.uploads[fileId]) {
+      const existingData = this.uploads[fileId];
+      const update = { uploadState: UploadStatus.UPLOAD_CANCELLED };
+      this.uploads[fileId] = { ...existingData, ...update };
+    }
+  }
+
+  performAbort(fileId: string) {
+    if (fileId in this.uploads) {
+      const existingData = this.uploads[fileId];
+      if (existingData?.abort) {
+        existingData?.abort?.();
```

Comment:
```suggestion
        existingData.abort();
```
is this optional check required here?

Reply:
Not required.